cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamFileJanitor.java

```diff
@@ -115,4 +120,12 @@ void clean(StreamConfig config, long currentTime) throws IOException {
   private boolean isPartitionDirector(Location location) throws IOException {
     return (location.isDirectory() && location.getName().indexOf('.') > 0);
   }
+
+  private boolean isStreamExists(Id.Stream streamId) throws IOException {
```

Comment:
I see. Do we care about other exceptions that ``StreamAdmin.exists`` may throw in other parts of the code? If not maybe we can have this cleanup in ``StreamAdmin.exists`` itself? May be in a different PR though.